src/plugins/GUITestBase/src/GTUtilsDashboard.cpp

```diff
@@ -35,6 +37,33 @@ using namespace HI;
 
 const QString GTUtilsDashboard::TREE_ROOT_ID = ExternalToolsDashboardWidget::TREE_ID;
 
+const QString GTUtilsDashboard::Notification::typeNames[] = {"INFO",
+                                                             "WARNING",
+                                                             "ERROR"};
+
+QString GTUtilsDashboard::Notification::typeToString(Type t) {
+    return typeNames[static_cast<int>(t)];
+}
+
+QVariant GTUtilsDashboard::Notification::stringToType(const QString &str) {
+    QVariant type;
+    if (str.isEmpty()) {
+        return type;
+    }
+
+    constexpr int typesNumber = sizeof(Notification::typeNames) / sizeof(QString);
+    for (int typeInd = 0; typeInd < typesNumber; typeInd++) {
+        if (QString::compare(str, Notification::typeNames[typeInd], Qt::CaseInsensitive) == 0) {
```

Comment:
What is the reason to define typeName values in the upper-case and next compare them case insensitive with notification types which are lower-case?

I would suggest to use lowercase map and replace typeNames array with QStringList.
In this case your method can be as simple as 1-line of code.
 1. typeNames.indexOf(str).